src/transactions/transactionBytes.js

```diff
@@ -12,181 +12,126 @@
+		lifetimeBuffer.writeInt8(multisigTransactionAsset.lifetime);
+		const keysgroupBuffer = Buffer.from(multisigTransactionAsset.keysgroup.join(''));
+
+		return Buffer.concat([minBuffer, lifetimeBuffer, keysgroupBuffer]);
 	}
 
 	/**
 	 * @method isDappTransaction
-	 * @return {object}
+	 * @return {Buffer}
 	 */
 
 	function isDappTransaction() {
 		const dapp = transaction.asset.dapp;
-		let buf = Buffer.from(dapp.name);
-
-		if (dapp.description) {
-			const descriptionBuf = Buffer.from(dapp.description);
-			buf = Buffer.concat([buf, descriptionBuf]);
-		}
-
-		if (dapp.tags) {
-			const tagsBuf = Buffer.from(dapp.tags);
-			buf = Buffer.concat([buf, tagsBuf]);
-		}
-
-		if (dapp.link) {
-			buf = Buffer.concat([buf, Buffer.from(dapp.link)]);
-		}
-
-		if (dapp.icon) {
-			buf = Buffer.concat([buf, Buffer.from(dapp.icon)]);
-		}
-
-		const bb = new ByteBuffer(4 + 4, true);
-		bb.writeInt(dapp.type);
-		bb.writeInt(dapp.category);
-		bb.flip();
-
-		buf = Buffer.concat([buf, bb.toBuffer()]);
-
-		return {
-			assetBytes: buf,
-			assetSize: buf.length,
-		};
+		const dappNameBuffer = Buffer.from(dapp.name);
+		const dappDescriptionBuffer = dapp.description ? Buffer.from(dapp.description) : Buffer.from('');
+		const dappTagsBuffer = dapp.tags ? Buffer.from(dapp.tags) : Buffer.from('');
+		const dappLinkBuffer = Buffer.from(dapp.link);
+		const dappIconBuffer = dapp.icon ? Buffer.from(dapp.icon) : Buffer.from('');
+		const dappTypeBuffer = Buffer.alloc(4);
+		dappTypeBuffer.writeInt8(dapp.type);
+		const dappCategoryBuffer = Buffer.alloc(4);
+		dappCategoryBuffer.writeInt8(dapp.category);
+
+		return Buffer.concat([
+			dappNameBuffer,
+			dappDescriptionBuffer,
+			dappTagsBuffer, dappLinkBuffer,
+			dappIconBuffer,
+			dappTypeBuffer,
+			dappCategoryBuffer,
+		]);
 	}
 
 	/**
 	 * @method isDappInTransferTransaction
-	 * @return {object}
+	 * @return {Buffer}
 	 */
 
 	function isDappInTransferTransaction() {
-		const buf = Buffer.from(transaction.asset.inTransfer.dappId);
-
-		return {
-			assetBytes: buf,
-			assetSize: buf.length,
-		};
+		return Buffer.from(transaction.asset.inTransfer.dappId);
 	}
 
 	/**
 	 * @method isDappOutTransferTransaction
-	 * @return {object}
+	 * @return {Buffer}
 	 */
 
 	function isDappOutTransferTransaction() {
-		const dappBuf = Buffer.from(transaction.asset.outTransfer.dappId);
-		const transactionBuf = Buffer.from(transaction.asset.outTransfer.transactionId);
-		const buf = Buffer.concat([dappBuf, transactionBuf]);
-
-		return {
-			assetBytes: buf,
-			assetSize: buf.length,
-		};
-	}
+		const dappOutAppIdBuffer = Buffer.from(transaction.asset.outTransfer.dappId);
+		const dappOutTransactionIdBuffer = Buffer.from(transaction.asset.outTransfer.transactionId);
 
-	/**
-	 * `transactionType` describes the available transaction types.
-	 *
-	 * @property transactionType
-	 * @type object
-	 */
+		return Buffer.concat([dappOutAppIdBuffer, dappOutTransactionIdBuffer]);
+	}
 
 	const transactionType = {
```

Comment:
This needs to be renamed, along with the functions for getting asset data for a given transaction type. Perhaps `assetDataGetters` and `getAssetDataForSendTransaction` etc?